core/src/main/java/io/syndesis/core/Tokens.java

```diff
@@ -38,13 +49,14 @@ public static String getAuthenticationToken() {
         Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
         if (authentication == null) {
             throw new IllegalStateException("Cannot set authorization header because there is no authenticated principal");
-        } else if (!KeycloakAuthenticationToken.class.isAssignableFrom(authentication.getClass())) {
+        }
```

Comment:
If you feel that's not related to this PR, that's true ;-) we can do it in a separate PR, too.